tests/test_rich_markup_mode.py

```diff
@@ -38,3 +41,186 @@ def main(arg: str):
 
     result = runner.invoke(app, ["--help"])
     assert any(c in result.stdout for c in rounded)
+
+
+@pytest.mark.parametrize(
+    "mode,lines",
+    [
+        ("markdown", ["First line", "Line 1 Line 2 Line 3", ""]),
+        ("rich", ["First line", "Line 1", "", "Line 2", "", "Line 3", ""]),
+        ("none", ["First line", "Line 1", "Line 2", "Line 3", ""]),
+    ],
+)
+def test_markup_mode_newline_pr815(mode: str, lines: List[str]):
```

Comment:
Here, the question is mostly about the newlines: should there be one after "First line" or not?

Further, should the `markdown` parser output "Line 1 Line 2 Line 3" as one string, or split them into multiple strings, similar to the current behaviour for the option `none`?

Reply:
Yep! In all 3 modes the lines *should* be:

* `First line`
* new line
* `Line 1`
* new line
* `Line 2`
* new line
* `Line 3`
* new line

```
["First line", "", "Line 1", "", "Line 2", "", "Line 3", ""]
```

I just tested with pure Rich:

```Python
from rich.console import Console
from rich.markdown import Markdown

console = Console()

content = """
First line

Line 1

Line 2

Line 3
"""

md = Markdown(content)


print("--- Markdown ---")

console.print(md)

print("--- Rich markup ---")

console.print(content, markup=True)

print("--- plain ---")

print(content)
```

And the output is almost always the same:

```console
$ python demo.py
--- Markdown ---
First line                                                                                                                                                                                                               

Line 1                                                                                                                                                                                                                   

Line 2                                                                                                                                                                                                                   

Line 3                                                                                                                                                                                                                   
--- Rich markup ---

First line

Line 1

Line 2

Line 3

--- plain ---

First line

Line 1

Line 2

Line 3


```

The main difference at least in this script is that for Markdown Rich doesn't add an extra line on top and below. But the main question we have right now are lines in between, and in all cases those are printed consistently.

So yep, that's a bug here in Typer.